pkg/webhook/pod/mutating/sidecarset.go

```diff
@@ -145,6 +149,22 @@ func (h *PodCreateHandler) sidecarsetMutatingPod(ctx context.Context, req admiss
 	return nil
 }
 
+func (h *PodCreateHandler) getSpecificHistorySidecarSet(sidecarSet *appsv1alpha1.SidecarSet) (*appsv1alpha1.SidecarSet, error) {
+	// if .Spec.InjectionStrategy.Revision is empty, just return the latest sidecarSet
+	if sidecarSet.Spec.InjectionStrategy.Revision == "" {
+		return sidecarSet.DeepCopy(), nil
+	}
+	// else return its corresponding history revision
+	hc := sidecarcontrol.NewHistoryControl(h.Client)
+	history, err := hc.GetHistorySidecarSet(sidecarSet, sidecarSet.Spec.InjectionStrategy.Revision)
+	if err != nil || history == nil {
```

Comment:
The failure in this place cannot be ignored and should be returned directly. Because the above are already matched pods, the failure here will only affect the matched pods, not the other business pods